New Azure TC: Add NIC after VM Provision

adriansuhov opened the pull request:
### Check if Network Interfaces are configured correctly in guest VM
This is an Azure only test. After the VM is provisioned, additional Network Interface(s) will be added. The number of Extra NICs is dynamic and is a paramater in the XML test definition (default is 2 Extra NICs in xml file). Steps:
1. Stop VM
2. Create NIC(s) inside the Resource Groups and attach them to the VM
3. Start VM
4. Check if IPs are assigned inside the VM for each new extra NIC

## Test Results - Ubuntu 18.04 is expected to fail
[LISAv2 Test Results Summary]
Test Run On           : 04/10/2019 09:21:16
ARM Image Under Test  : Canonical : **UbuntuServer : 18.04-LTS : latest**
Total Test Cases      : 1 (0 Passed, **1 Failed**, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:6

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-ADD-NIC-AFTER-PROVISION                                                       FAIL                 4.27

[LISAv2 Test Results Summary]
Test Run On           : 04/10/2019 09:47:53
ARM Image Under Test  : **Canonical : UbuntuServer : 16.04-LTS : latest**
Total Test Cases      : 1 (**1 Passed,** 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:32

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-ADD-NIC-AFTER-PROVISION                                                       PASS                24.41

[LISAv2 Test Results Summary]
Test Run On           : 04/10/2019 09:31:29
ARM Image Under Test  : **OpenLogic : CentOS : 7.6 : 7.6.20181219**
Total Test Cases      : 1 (**1 Passed**, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-ADD-NIC-AFTER-PROVISION                                                       PASS                 4.08

[LISAv2 Test Results Summary]
Test Run On           : 04/10/2019 08:09:58
ARM Image Under Test  : **RedHat : RHEL : 7.5 : latest**
Total Test Cases      : 1 (**1 Passed,** 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-ADD-NIC-AFTER-PROVISION                                                       PASS                 4.05

[LISAv2 Test Results Summary]
Test Run On           : 04/10/2019 08:41:34
ARM Image Under Test  : **SUSE : SLES : 15 : latest**
Total Test Cases      : 1 (**1 Passed**, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:11

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 NETWORK              NET-ADD-NIC-AFTER-PROVISION                                                       PASS                  3.6

lizzha commented on Documents/Release-model.md:
```diff
@@ -0,0 +1,13 @@
+
+## Objective and steps
+
+This document describes about LISav2 branch management and its release procedure to Jenkins server. Release request can be made by time or feature base.
```
typo: LISav2 -> LISAv2

ader1990 replied:
Why not: "This document describes LISAv2 branch management and its release procedure. A release request can be made by time or feature base."?

And for the record, what is a release request? Are you referring to a release itself?

juhlee-microsoft replied:
@lizzha Fixed.

@ader1990 I recall our discussion about when we can release. It could be about feature-release or bug-fix for distro partner. Regarding feature-release, it should be on-demand style release. If the bug is critical, then it is also on-demand style. Otherwise, we can release bug-fixes in monthly release.

Omitted 'Jenkins', but mostly we will do menu refresh there.

A release is simple tag in our model not 'Release' archive in this repo.

175422bb9147e378282355d35d4471e7c7d272db

lizzha commented on Documents/Release-model.md:
```diff
@@ -0,0 +1,13 @@
+
+This document describes about LISav2 branch management and its release procedure to Jenkins server. Release request can be made by time or feature base.
+
+LISAv2 master branch will merge the changes from feature branches. Based on standard software practice, festure branches will be merged to master via PR review process. Up on request of release schedule, it will have a new tag in master branch. After passing all release readiness requirements including full validation and review, the code will deploy to Jenkins server.
```
typo: festure branches will be -> feature ...

ader1990 replied:
The content of this paragraph is a little confusing. Why not
>> "The changes from feature branches will be merged to LISAv2 *master* branch. Based on standard software practice, festure branches will be merged to master via PR review process. Upon request of a release schedule, 
a new tag based on master branch will be created. After passing all release readiness requirements including full validation and review, the code will be used by the production Jenkins pipelines."

juhlee-microsoft replied:
@lizzha Fixed

@ader1990 Updated with yours

175422bb9147e378282355d35d4471e7c7d272db

lizzha commented on Documents/Release-model.md:
```diff
@@ -0,0 +1,13 @@
+
+LISAv2 master branch will merge the changes from feature branches. Based on standard software practice, festure branches will be merged to master via PR review process. Up on request of release schedule, it will have a new tag in master branch. After passing all release readiness requirements including full validation and review, the code will deploy to Jenkins server.
+
+![alt text](https://github.com/LIS/LISAv2/blob/juhlee/release_model_doc/Documents/LISAv2ReleaseDiagram.jpg)
```
need to change the branch to master

juhlee-microsoft replied:
@lizzha Clarification or which line number?

lizzha replied:
I mean the link of the image file

juhlee-microsoft replied:
Fixed 536890fd846467b007cc5779aa062e4cc37cbb1c

ader1990 commented:
Squash the commits to only one and provide a relevant commit message like:
docs: Added LISAv2 release model

chvalean commented on Documents/Release-model.md:
```diff
@@ -0,0 +1,13 @@
+
+This document describes about LISav2 branch management and its release procedure to Jenkins server. Release request can be made by time or feature base.
+
+LISAv2 master branch will merge the changes from feature branches. Based on standard software practice, festure branches will be merged to master via PR review process. Up on request of release schedule, it will have a new tag in master branch. After passing all release readiness requirements including full validation and review, the code will deploy to Jenkins server.
```
should be "Upon" not "Up on".
Also agree with Adrian's version.

juhlee-microsoft replied:
@chvalean Fixed 175422bb9147e378282355d35d4471e7c7d272db

adriansuhov committed:
New Azure TC: Add NIC after Provision

This is an Azure only test. After the VM is provisioned, additional
Network Interface(s) will be added. The number of Extra NICs is dynamic
and is a paramater in the XML test definition. Steps:
1. Stop VM
2. Create NIC(s) inside the Resource Groups and attach them to the VM
3. Start VM
4. Check if IPs are assigned inside the VM for each new extra NIC

juhlee-microsoft committed:
Merge branch 'master' into master

juhlee-microsoft committed:
Merge branch 'master' into master
